pytorch-cpu-requirements.txt

```diff
@@ -1,4 +1,4 @@
 --index-url https://download.pytorch.org/whl/cpu
-torch>=2.3.0,<2.6.0
+torch>=2.3.0,<=2.7.1
```

Comment:
Please tag https://github.com/iree-org/iree-turbine/issues/348 in the PR description. Seems like we can call that fixed with this PR?

What is limiting this to <=2.7.1 vs going further ahead? Looks like that is just the latest version released now, so this will insulate us from new releases breaking things? Could put that information in the PR description too.

We could also have test workflows use a matrix across torch versions... and maybe a nightly/scheduled job testing unpinned torch so we get early notifications of failures (or have dependabot try updating after new versions are released).

Reply:
dropped 2.3 and 2.4